src/passkey_child/passkey_child_credentials.c

```diff
@@ -389,7 +394,14 @@ print_credentials(const struct passkey_data *data,
         goto done;
     }
 
-    printf("passkey:%s,%s\n", b64_cred_id, pem_key);
+    b64_user_id = sss_base64_encode(tmp_ctx, data->user_id, USER_ID_SIZE);
+    if (b64_user_id == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "failed to encode user id.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    printf("passkey:%s,%s,%s\n", b64_cred_id, pem_key, b64_user_id);
```

Comment:
Hi,

I wonder if it would make sense to only print the `b64_user_id` for discoverable credentials. Since it will not be in the assertion for server-side credentials, it will be useless when stored on the IPA server. Additionally the IPA server would have to store if the credential is discoverable or not, i.e. if the user id can be expected in the assertion or not. If the `b64_user_id` is not printed for server-side credential the IPA server can already see from the number of elements (2 or 3) in the string what is the credential type and if a user id is expected in the assertion or not.

@flo-renaud what do you think would make more sense on the IPA server side?

bye,
Sumit

Reply:
@sumit-bose 
With your proposal IPA would have to support 2 different formats:
- for server-side credentials: `passkey:cred_id,pem_key`
- for discoverable credentials: `passkey:cred_id,pem_key,user_id`

That's not an issue, IPA just needs to adapt the method that is validating the format. And as you wrote, it allows to easily distinguish between the 2 types of credentials.

Reply:
Chiming in here to ask about the intentions with regard to User ID, see questions from previous thread: https://github.com/SSSD/sssd/pull/6453#discussion_r1031145013

Reply:
> Chiming in here to ask about the intentions with regard to User ID, see questions from previous thread: [#6453 (comment)](https://github.com/SSSD/sssd/pull/6453#discussion_r1031145013)
> 
> Reason for asking is that the random User ID per credential was fine for server-side credentials, but may be problematic in terms of credential management of discoverable credentials on the authenticator.

Hi,

the idea is to have an additional attribute which is only known by the KDC and the authentication device and never send over the network during authentication. If the KDC receives the assertion data for validation the presence of the user id can be used as additional confirmation that the expected device was used.

HTH

bye,
Sumit

Reply:
> the idea is to have an additional attribute which is only known by the KDC and the authentication device and never send over the network during authentication. 

Could you elaborate what you mean here? Wouldn't the User ID have to be sent over the network for the KDC to validate it?

> If the KDC receives the assertion data for validation the presence of the user id can be used as additional confirmation that the expected device was used.

WebAuthn does this too, but to confirm the expected user or to identify a user (for "username"-less logins) rather than authenticator. A valid assertion signature provides strong proof that the user (authenticator) holds the corresponding private key.

While it's perfectly possible to assign a new randomized User ID to each registered credential, it might be better UX if the same User ID is re-used each time. This overwrites previously held discoverable credentials on the authenticator and simplifies credential management, especially if the user wants to manage credentials themselves (e.g. `ykman fido credentials list|delete`). It was my understanding (per the previously linked thread) that a per-user (rather than per-credential) User ID was the long-term goal. 

Either way, just my 2 cents, apologies for the slight off-topic comments. :-)